compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/QueryDesugar.java

```diff
@@ -1905,17 +1905,17 @@ public void visit(BLangSimpleVarRef bLangSimpleVarRef) {
         BSymbol symbol = bLangSimpleVarRef.symbol;
         String identifier = bLangSimpleVarRef.variableName == null ? String.valueOf(bLangSimpleVarRef.varSymbol.name) :
                 String.valueOf(bLangSimpleVarRef.variableName);
-        BSymbol resolvedSymbol = symResolver.lookupClosureVarSymbol(env,
-                Names.fromString(identifier), SymTag.VARIABLE);
+        if (symbol instanceof BVarSymbol) {
+            BVarSymbol originalSymbol = ((BVarSymbol) symbol).originalSymbol;
+            if (originalSymbol != null) {
+                symbol = originalSymbol;
+            }
+        }
```

Comment:
Instead of using `instanceof`, We can use `kind` of the symbol for the this check.
```java
if (symbol.getKind() == SymbolKind.VARIABLE) {
    BVarSymbol originalSymbol = ((BVarSymbol) symbol).originalSymbol;
    if (originalSymbol != null) {
        symbol = originalSymbol;
    }
}
```

Reply:
Also, make sure to check if the `symbol` is null at the beginning and return if it is.
```java
if (symbol == null) {
    result = bLangSimpleVarRef;
    return;
}
```